rbac/management/relation_replicator/relations_api_check.py

```diff
@@ -0,0 +1,223 @@
+
+        res = connection.getresponse()
+        data = res.read()
+        json_data = json.loads(data)
+
+        token = json_data["access_token"]
+        return token
+
+
+class JWTManager:
+    """Class to handle management of JWT tokens."""
+
+    def __init__(self, jwt_provider, jwt_cache):
+        """Establish connection to JWT cache and provider."""
+        self.jwt_cache = jwt_cache
+        self.jwt_provider = jwt_provider
+
+    def get_jwt_from_redis(self):
+        """Retrieve jwt token from redis or generate from Redhat SSO if not exists in redis."""
+        try:
+            # Try retrieve token from redis
+            token = self.jwt_cache.get_jwt_response()
+
+            # If token not is redis
+            if not token:
+                token = self.jwt_provider.get_jwt_token(
+                    settings.RELATIONS_API_CLIENT_ID, settings.RELATIONS_API_CLIENT_SECRET
+                )
+                # Token obtained store it in redis
+                if token:
+                    self.jwt_cache.set_jwt_response(token)
+                    logger.info("Token stored in redis.")
+                else:
+                    logger.error("Failed to store jwt token in redis.")
+            else:
+                # Token exists return it
+                logger.info("Token retrieved from redis.")
+            return token
+
+        except Exception as e:
+            logger.error(f"error occurred when trying to retrieve JWT token. {e}")
+            return None
+
+
+jwt_cache = JWTCache()
+jwt_provider = JWTProvider()
+jwt_manager = JWTManager(jwt_provider, jwt_cache)
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@contextmanager
+def create_client_channel(addr):
+    """Create secure channel for grpc requests."""
+    secure_channel = grpc.insecure_channel(addr)
+    yield secure_channel
+
+
+class RelationsApiAssignmentCheck(RelationReplicator):
+    """Checks relations via the Relations API over gRPC."""
+
+    def replicate(self, event: ReplicationEvent):
+        """Replicate the given event to Kessel Relations via the gRPC API."""
+        assignments = self._check_relationships(relationships=event.add)
+        return assignments
+
+    def _check_relationships(self, relationships):
+        relations_assignments = {"group_uuid": "", "principal_relations": []}
+        for r in relationships:
+            relation_exists = check_relation_core(r)
+            relations_assignments["group_uuid"] = r.resource.id
+            relations_assignments["principal_relations"].append(
+                {"id": r.subject.subject.id, "relation_exists": relation_exists}
+            )
+            if not relation_exists:
+                logger.warning(
+                    f"Relation missing: User ID {r.subject.subject.id} is not associated with Group ID {r.resource.id}"
+                )
+        return relations_assignments
+
+
+def check_relation_core(r: common_pb2.Relationship) -> bool:
```

Comment:
This method can go to `RelationsApiAssignmentCheck`. We will see later if we will need this method anywhere else.